### DISCUSSION TITLE
The size of tensor a (41) must match the size of tensor b (13) at non-singleton dimension 3

### DISCUSSION THREAD

**Discussion by luoMonkeyKing** [openai/whisper#2172] [Q&A]
Does whisper not support multi-threaded concurrent calls? The global variable I use when loading the model: ```whisper.load_model(name="large", download_root='/root').to(device)```, initialize the instance in the thread:
```torch.hub.load(repo_or_dir='/root/snakers4_silero-vad_master',model='silero_vad',force_reload=False,trust_repo=True,source='local',onnx=False)```
But when using multi-trip concurrent call whisper, the following error is reported:
```RuntimeError: The size of tensor a (41) must match the size of tensor b (13) at non-singleton dimension 3```
If whisper.load_model is loaded in a thread pool, concurrency can be supported, but this is too slow. The model needs to be loaded every time. Is there any better way to support multiple concurrency?
Flask + Python 3.11

**Comment by glangford** [comment]
The whisper API is not thread safe.

* https://github.com/openai/whisper/discussions/296#discussioncomment-3854090
